### DISCUSSION TITLE
Can't Define a new Raspberry pi4 as a child

### DISCUSSION THREAD

**Discussion by DearNebReddey** [wehr-lab/autopilot#143] [Q&A] [answered]
After the installation, i launched the setup with (python3 -m autopilot.setup) command and when I try to setup this raspberry as a child, I got this error message.

![defineachilderror](https://user-images.githubusercontent.com/92102377/141087336-38aec6b4-f1f6-4012-8d24-b8212bf8dde4.png)

So should I Define it as pilot even if I have a pilot and terminal defined already ? , it will be considered as a second pilot or it will ignore the defined one and it will consider it as the first and only pilot ? 

Thank you

**Comment by cxrodgers** [comment, marked as answer]
Thanks for posting. I have noticed this bug and hadn't gotten around to posting it yet, sorry. Yes, define it as a second Pilot, not as a Child. Then, on the next few screens, you have the option to set the PARENTID and PARENTIP to be the name and ip of the parent Pilot, respectively. That is how it "knows" it's a Child.

Also, during the setup of the Parent Pi, you can tell it the name of the Child in the field CHILDID. You can change this directly in prefs.json as well.

I think we need to remove that initial choice to "setup as Child", since I think it just generates the error message you posted. 

In response to your other question, the Terminal should be completely fine with having multiple Pilots connected, it doesn't only take the first one or anything like that.

**Reply by sneakers-the-rat** [reply]
Yes, the notion of a "child" should be deprecated altogether, it turned out to not be necessary and just causes a lot of problems -- having no sense of hierarchy among the pis seems to work just fine.

**Reply by DearNebReddey** [reply]
Thank you for your answers, next week I will try to implement the Real-time DeepLabCut example on the Quick-start section of the Docs but with a USB Camera, so maybe there will be some questions in the way, Thanks

**Reply by sneakers-the-rat** [reply]
I'll try to be ready for them! Still working out dependency management for something that was born for a plugin structure but only developed it recently. you'll need to make sure you have both of the deeplabcut and dlc-live packages installed on whatever you're using for the processing computer, and probably opencv (`python -m autopilot.setup.run_script opencv`), but i have used at least the dlc-live module recently and had no problems. sorry for the lack of test coverage! am stretched thin.
